pkg/controllers/provisioning/provisioner.go

```diff
@@ -118,7 +119,7 @@ func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (resul
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !p.cluster.Synced(ctx) {
-		logging.FromContext(ctx).Debugf("waiting on cluster sync")
+		log.FromContext(ctx).V(1).Info("waiting on cluster sync")
```

Comment:
@jonathan-innis  thoughts on making this v(2)? This log itself is the reason I often times turn off debug logs. This and other cluster sync logs show up quite a bit. 

We can see here the cluster sync logs are double that of the rest of the logs. 

~ » k logs -n karpenter karpenter-6b4b7bfddb-dv6nt | grep "cluster sync" | wc -l                                        
   15937

~ » k logs -n karpenter karpenter-6b4b7bfddb-dv6nt | grep -v "cluster sync" | wc -l                                     
    9102

Reply:
@Bryce-Soghigian @jonathan-innis 


I've structured the messages as follows: I've used `V(0)` for all previous `Info` messages and `V(1)` for all previous `Debugf` messages. By default, zap doesn't display `V(1)` level messages, but it can be enabled if necessary. For example, with my current patch, users won't see all the `V(1)` level messages. 

**1. Default logging behavior:** 
  ```Go
  func init() {
	  logger := zap.New()
	  log.SetLogger(logger)
  }
  func main() {
	  log.FromContext(context.Background()).V(0).Info("Logged with a verbosity level of 0")
	  log.FromContext(context.Background()).V(1).Info("Logged with a verbosity level of 1")
	  log.FromContext(context.Background()).V(2).Info("Logged with a verbosity level of 2")
  }
```
This prints the following:
```
{"level":"info","ts":"2024-03-07T11:38:16+02:00","msg":"Logged with a verbosity level of 0"}
```

**2. When we want to enable the `V(1)` level messages, it looks like following:**
```Go
func init() {
	logger := zap.New(zap.Level(zapcore.Level(-1)))
	log.SetLogger(logger)
}
func main() {
	log.FromContext(context.Background()).V(0).Info("Logged with a verbosity level of 0")
	log.FromContext(context.Background()).V(1).Info("Logged with a verbosity level of 1")
	log.FromContext(context.Background()).V(2).Info("Logged with a verbosity level of 2")
}
```
This prints the following:
```
{"level":"info","ts":"2024-03-07T11:40:40+02:00","msg":"Logged with a verbosity level of 0"}
{"level":"debug","ts":"2024-03-07T11:40:40+02:00","msg":"Logged with a verbosity level of 1"}
```

Perhaps we need some decisions about
1. Should we include debug messages (`V(1)`) for printing? Are debug messages more beneficial for users or developers or both?
2. Is it necessary to conduct a thorough scan of the code base to find if any `Info` messages should be converted to debug and vice versa? (IMO, this doesn't need to be addressed in this pull request and can be done in a follow-up)?

Reply:
> thoughts on making this v(2)? This log itself is the reason I often times turn off debug logs. This and other cluster sync logs show up quite a bit

I'd personally not like to expand out of two log levels (three if you count error) at this point without having a bunch of examples of what should be "trace"-level logs and an explicit definition of what should go into trace and what should go into debug.

Without having a clear dileniation here, it's going to be really tough for new contributors (or even existing contributors) to figure out whether something should be trace, debug, or info. I have enough trouble differentiating these things today. I'm curious how spammy "waiting on cluster sync" is today in your debug logs

Reply:
> . I'm curious how spammy "waiting on cluster sync" is today in your debug logs

It makes up over 2/3 of our total log volume.

Reply:
Oh, wow. Yeah, that is a lot. I'd rather just remove it than anything else if it's spammy right now. We have metrics around this now so I'm skeptical that we actually need a log to represent this if someone can just look at a metric to see that their cluster state isn't synced. The other thing that you could do here is define a time which generally we see the cluster state syncing. If we get past that time, only then do we log a message. That way users have insights into why this is hanging, but we aren't spamming the logs all of the time. 

@Bryce-Soghigian I'd like to decouple this from this PR if we can. Though, I'd agree that we should improve here. Can you open an issue (and open the PR to fix it if you are willing to) to improve this logging for this spammy message?